### PR TITLE
Adds support for case insensitivity in field names

### PR DESCRIPTION
**Pull request by desaikd**
## Description
This PR adds support for case insensitivity in field names for Ion parser and makes it consistent with json parser.

## List of changes
- [Adds support for case-insensitivity for field names](https://github.com/rmarrowstone/trino/commit/a672b42c1d52c335ca2b1bcb01beeeb905fc4c95)
   - Converts the field name into lower case for both Decoder and Encoder to ensure case-insensitivity for field names/keys
- [Adds tests for case insensitivity in field names](https://github.com/rmarrowstone/trino/commit/c2186e8e4abebd088c0dc1f0c6e117295ee56ce8)

### CODE REVIEW

File: lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonDecoderFactory.java [hunk 1]
```diff
@@ -71,7 +72,7 @@ public static IonDecoder buildDecoder(List<Column> columns)
     {
         return RowDecoder.forFields(
                 columns.stream()
-                        .map(c -> new RowType.Field(Optional.of(c.name()), c.type()))
+                        .map(c -> new RowType.Field(Optional.of(c.name().toLowerCase(Locale.ROOT)), c.type()))
```
Comment (rmarrowstone) [inline lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonDecoderFactory.java]: Is this necessary? AFAICT the lowercasing in `RowDecoder.forFields()` is sufficient.

File: lib/trino-hive-formats/src/test/java/io/trino/hive/formats/ion/TestIonFormat.java [hunk 2]
```diff
@@ -65,6 +65,32 @@ public void testSuperBasicStruct()
...
+                        field("BAR", VARCHAR)),
+                "{ Bar: baz, foo: 31 }",
+                List.of(31, "baz"));
+    }
+
+    @Test
+    public void testCaseInsensitivityOfDuplicateKeys()
+            throws IOException
+    {
+        // this test asserts that duplicate key behavior works as expected(i.e. capturing the last value),
```
Comment (rmarrowstone) [inline lib/trino-hive-formats/src/test/java/io/trino/hive/formats/ion/TestIonFormat.java:84]: nice edge case identification!

**Review by rmarrowstone — APPROVED** [review]

File: lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonDecoderFactory.java (hunk 1 above)
Comment (desaikd) [reply lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonDecoderFactory.java]: Thats right, `RowDecoder.forFields` already performs lower casing. Will remove it from here.
